### PR TITLE
add `workingDirectory` support to `eslint` client

### PR DESCRIPTION
**Pull request by mpsq**
Logic from
https://github.com/microsoft/vscode-eslint/blob/main/client/src/client.ts#L642.
Pattern support is not implemented in this change.

Sorry in advance for the sub-optimal code, I am not very proficient with elisp!

See #3438

### CODE REVIEW

File: clients/lsp-eslint.el [hunk 1]
```diff
@@ -260,27 +260,45 @@ stored."
...
+                              :nodePath lsp-eslint-node-path
+                              :workingDirectory (when working-directory
+                                                  (list
+                                                   :directory working-directory
+                                                   :!cwd :json-false))
+                              :workspaceFolder (list :uri (lsp--path-to-uri workspace-folder)
+                                                     :name (f-filename workspace-folder))))))))
        (apply #'vector)))
 
+(defun lsp--working-directory (workspace current-file)
```
Comment (yyoncho) [inline clients/lsp-eslint.el:289]: can you prefix with `lsp-eslint--`? Can you also drop one line what this method is supposed to do.

**Review by yyoncho — COMMENTED** [review]
Looks good, one minor nit.

File: clients/lsp-eslint.el (hunk 1 above)
Comment (mpsq) [reply clients/lsp-eslint.el]: Done

File: clients/lsp-eslint.el [hunk 2]
```diff
@@ -260,27 +260,45 @@ stored."
...
+  (let* ((found)
+         (i 0))
+    (while (and (< i (length lsp-eslint-working-directories))
+                (not found))
+      (let ((res (lsp-resolve-value (elt lsp-eslint-working-directories i))))
+        (when (not (f-absolute? res))
+          (setq res (f-join workspace res)))
+        (if (string-match-p res current-file)
```
[the remark this answers is not in the document]
Comment (mpsq) [reply clients/lsp-eslint.el]: Done

**Comment by mpsq** [comment]
@yyoncho anything I can do / help with to get this PR merged?

**Comment by yyoncho** [comment]
No, sorry for the delay.

**Pull request merged**